client/wfm/src/service/WfmService.ts

```diff
@@ -0,0 +1,241 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as shortid from 'shortid';
+import { WorkOrderResult } from '../result/WorkOrderResult';
+import { STATUS } from '../status';
+import { Step } from '../step/Step';
+import { WorkFlow } from '../workflow/WorkFlow';
+import { WorkOrder } from '../workorder/WorkOrder';
+import { DataService } from './DataService';
+import { ResultService } from './ResultService';
+import { UserService } from './UserService';
+
+interface CompleteStepParams {
+  /** The ID of the workorder to complete the step for */
+  workorderId: string;
+  /** The submission to save */
+  submission: object;
+  /** The ID of the step to save the submission for */
+  stepCode: string;
+}
+
+export class WfmService {
+
+  constructor(
+    protected workorderService: DataService<WorkOrder>,
+    protected workflowService: DataService<WorkFlow>,
+    protected resultService: ResultService,
+    protected userService: UserService
+  ) {
+  }
+
+  /**
+   * Beginning a workflow for a single Workorder.
+   *
+   * @param {string} workorderId - The ID of the workorder to begin the workflow for.
+   */
+  public beginWorkflow(workorderId) {
+    return this.workorderSummary(workorderId).then(summary => {
+      const workorder = summary.workorder;
+      const workflow = summary.workflow;
+      const res = summary.result || this.createNewResult(workorderId, workorder.assignee);
+
+      // When the result has been read/created, then we can move on.
+      return Promise.resolve(res).then(result => {
```

Comment:
We were saying that framework code should be really callback based and we can wrap things into promises in clients. This will bring some inconsistence, comparing to the other modules we have at the moment. 

Do you think we should made exception for client side modules ?